api/src/test/java/com/epam/pipeline/manager/docker/DockerRegistryNotificationTest.java

```diff
@@ -16,305 +16,374 @@
+        doReturn(getClonedTool(tool1, TOOL_ID_1)).when(mockToolManager).create(eq(tool1), eq(false));
+    }
+
+    private void setMockBehaviour2() {
+        setCommonBehaviour(registry1, REPO, IMAGE_NEW_GROUP, null, false);
+        doReturn(Optional.empty()).when(mockToolManager).loadToolInGroup(eq(IMAGE_NEW_GROUP), eq(null));
+        doReturn(getClonedTool(tool2, TOOL_ID_2)).when(mockToolManager).create(eq(tool2), eq(false));
+    }
+
+    private void setMockBehaviour4() {
+        setCommonBehaviour(registry1, REPO, IMAGE, toolGroup1, true);
+        doReturn(Optional.empty()).when(mockToolManager).loadToolInGroup(eq(IMAGE), eq(TOOLGROUP_ID_1));
+        doReturn(getClonedTool(tool1, TOOL_ID_1)).when(mockToolManager).create(eq(tool1), eq(false));
+    }
+
+    private void setMockBehaviour5() {
+        setCommonBehaviour(registry2, EXTERNAL_PATH, IMAGE, null, false);
+        doReturn(Optional.empty()).when(mockToolManager).loadToolInGroup(eq(IMAGE), eq(null));
+        doReturn(getClonedTool(tool3, TOOL_ID_3)).when(mockToolManager).create(eq(tool3), eq(false));
+    }
+
+    private void setMockBehaviour6() {
+        setCommonBehaviour(registry3, REPO_NO_WRITE_ACCESS, IMAGE_NEW_GROUP, null, false);
+        doReturn(NOT_ALLOWED_MESSAGE).when(mockMessageHelper).getMessage(
+                eq(MessageConstants.ERROR_PERMISSION_IS_NOT_GRANTED),
+                eq(REPO_NO_WRITE_ACCESS), eq(AclPermission.WRITE_PERMISSION));
+    }
+
+    private void setMockBehaviour7() {
+        setMockBehaviour1();
+        doReturn(Optional.of(getClonedTool(tool1, TOOL_ID_1)))
+                .when(mockToolManager).loadToolInGroup(eq(IMAGE), eq(TOOLGROUP_ID_1));
+    }
+
+    private void setCommonBehaviour(final DockerRegistry registry, final String repo, final String image,
+                                    final ToolGroup toolGroup, final boolean doesToolGroupExist) {
+        if (EXTERNAL_PATH.equals(repo)) {
+            doReturn(registry).when(mockDockerRegistryDao).loadDockerRegistryByExternalUrl(eq(EXTERNAL_PATH));
+        } else {
+            doReturn(registry).when(mockDockerRegistryDao).loadDockerRegistry(eq(repo));
+        }
+        doReturn(false).when(mockMetadataManager).hasMetadata(
+                eq(getEntityVO(registry.getId(), AclClass.DOCKER_REGISTRY)));
+        doReturn(splitAndGetPairOf(image)).when(mockToolGroupManager).getGroupAndTool(eq(image));
+        doReturn(doesToolGroupExist).when(mockToolGroupManager).doesToolGroupExist(eq(repo), eq(getGroupFrom(image)));
+        doReturn(getUserContext(CONTEXT_ID, USER, ROLE_ID, ROLE_USER)).when(mockUserManager).loadUserContext(eq(USER));
+        if (toolGroup != null) {
+            doReturn(toolGroup).when(mockToolGroupManager).loadByNameOrId(eq(getIdentifierFrom(registry1, IMAGE)));
+        }
+    }
+
+    private ImmutablePair<String, String> splitAndGetPairOf(String string) {
```

Comment:
To my mind, at the test classes, it would be cleaner to have utility constants instead of utility methods. Could you please consider getting rid of such methods like `splitAndGetPairOf`, `getIdentifierFrom`, `getGroupFrom`?

Reply:
Done!